owncloud/core
# [stable10] Backport of Make email field as default to create user

Make email field as default to create user.
Previously we had username, password. With
this change it becomes username, email. This
would help admins to create users with email
address.

Signed-off-by: Sujith H <sharidasan@owncloud.com>



## Description

Make email field as default to create a user. The old option of creating user with username and password is still available in the settings.

## Related Issue




- Fixes <issue_link>

## Motivation and Context

Make email field as default to create a user. The old option of creating user with username and password is still available in the settings.

## How Has This Been Tested?



- Create `admin` user and navigate to users page.
- [x] By default user should see following text box `Username`, `E-Mail` ( Previously it was `Username`, `Password`)
- The password option is still available. The user has to click the settings button available in the left bottom of the page and enable `Set password for new users`
- Once the email is configured in the config.php, create a new user with Username and E-Mail.
- An email should arrive to the user saying account is created.
- Click on the link provided in the email to set the password. The user should  be able to set the password using the link provided.
- The message that arrives during token expiry is verified.
- The message that arrives when token is invalid or token mismatch happens is also verified.
- When the token is expired a page showing resend activation link button appears. Which when clicked another email is sent to the user. The user needs to click the link from the email to set the password.
- Adjusted the unit tests for the backport.
- Verified with the previous setup of `Username` + `Password`. It worked.

## Screenshots (if appropriate):

## Types of changes


## Checklist:



## Open tasks:
